tests/integration/models/multifidelity/test_multifidelity_models.py

```diff
@@ -0,0 +1,574 @@
+    lf_prediction_mean, lf_prediction_var = (
+        concat_prediction_mean[:60],
+        concat_prediction_var[:60],
+    )
+
+    (
+        lf_prediction_direct_mean,
+        lf_prediction_direct_var,
+    ) = model.lowest_fidelity_signal_model.predict(test_locations_30)
+
+    npt.assert_array_equal(lf_prediction_mean, lf_prediction_direct_mean)  # , rtol=1e-7)
+    npt.assert_array_equal(lf_prediction_var, lf_prediction_direct_var)  # , rtol=1e-7)
+
+
+def test_multifidelity_autoregressive_predict_hf_is_consistent_when_rho_zero() -> None:
+
+    xs_low = tf.Variable(np.linspace(0, 10, 100), dtype=tf.float64)[:, None]
+    xs_high = tf.Variable(np.linspace(0, 10, 10), dtype=tf.float64)[:, None]
+    lf_obs = tf.sin(xs_low)
+    hf_obs = 2 * tf.sin(xs_high) + tf.random.normal(
+        xs_high.shape, mean=0, stddev=1e-1, dtype=tf.float64
+    )
+
+    lf_query_points = add_fidelity_column(xs_low, 0)
+    hf_query_points = add_fidelity_column(xs_high, 1)
+
+    lf_dataset = Dataset(lf_query_points, lf_obs)
+    hf_dataset = Dataset(hf_query_points, hf_obs)
+
+    dataset = lf_dataset + hf_dataset
+
+    search_space = Box([0.0], [10.0])
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(
+            dataset, num_fidelities=2, input_search_space=search_space
+        )
+    )
+
+    model.update(dataset)
+    model.optimize(dataset)
+
+    model.rho[1] = 0.0  # type: ignore
+
+    test_locations = tf.Variable(np.linspace(0, 10, 32), dtype=tf.float64)[:, None]
+    hf_test_locations = add_fidelity_column(test_locations, 1)
+
+    hf_prediction = model.predict(hf_test_locations)
+    hf_prediction_direct = model.fidelity_residual_models[1].predict(test_locations)
+
+    npt.assert_array_equal(hf_prediction, hf_prediction_direct)
+
+
+def test_multifidelity_autoregressive_predict_hf_is_consistent_when_lf_is_flat() -> None:
+
+    xs_low = tf.Variable(np.linspace(0, 10, 100), dtype=tf.float64)[:, None]
+    xs_high = tf.Variable(np.linspace(0, 10, 10), dtype=tf.float64)[:, None]
+    lf_obs = tf.sin(xs_low)
+    hf_obs = 2 * tf.sin(xs_high) + tf.random.normal(
+        xs_high.shape, mean=0, stddev=1e-1, dtype=tf.float64
+    )
+
+    lf_query_points = add_fidelity_column(xs_low, 0)
+    hf_query_points = add_fidelity_column(xs_high, 1)
+
+    lf_dataset = Dataset(lf_query_points, lf_obs)
+    hf_dataset = Dataset(hf_query_points, hf_obs)
+
+    dataset = lf_dataset + hf_dataset
+
+    search_space = Box([0.0], [10.0])
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(
+            dataset, num_fidelities=2, input_search_space=search_space
+        )
+    )
+
+    model.update(dataset)
+    model.optimize(dataset)
+
+    kernel = gpflow.kernels.Matern52(variance=1e-18, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(dataset.astuple(), kernel, noise_variance=1e-5)
```

Comment:
This is weird, it should crash, not return a flat model. 

Besides your `dataset` contains a fidelity column?

Maybe instead keep a kernel variance of one and make all your observations in `lf_dataset` equal to zero? Prediction variance will not be zero but it doesn't matter for this test unless I'm mistaken.